src/ir/template.rs

```diff
@@ -169,14 +169,8 @@ pub trait TemplateParameters {
         );
 
         let id = *self.as_ref();
-        ctx.resolve_item(id).all_template_params(ctx).map(
-            |all_params| {
-                all_params
-                    .into_iter()
-                    .filter(|p| ctx.uses_template_parameter(id, *p))
-                    .collect()
-            },
-        )
+        ctx.resolve_item(id).all_template_params(ctx)
```

Comment:
Hmm, in particular, `used_template_params` used to mean a different thing with the `Option<..>`...

It used to return `Some` if there was any (used or unused) template parameter. then the `Vec` inside contained only the ones that were actually used...